net/src/ipc/spawn.rs

```diff
@@ -123,16 +140,30 @@ pub fn ipc_spawn(
 
     // close the pipe since we can never read from it again...
     child.stdout = None;
-
-    log_i!("READY! {} {:?}", out.ipc_binding, out.p2p_bindings);
+    log_i!(
+        "READY! {} {:?} [{}]",
+        out.ipc_binding,
+        out.p2p_bindings,
+        real_pid
+    );
 
     // Set shutdown function to kill the sub-process
     out.kill = Some(Box::new(move || {
         match child.kill() {
-            Ok(()) => (),
+            Ok(()) => kill_child(&real_pid),
             Err(e) => println!("failed to kill ipc sub-process: {:?}", e),
         };
     }));
 
     Ok(out)
 }
+
+#[cfg(windows)]
+fn kill_child(pid: &str) {
+    let mut child_killer = std::process::Command::new("taskkill");
+    child_killer.args(&["/pid", pid, "/f", "/t"]);
+    let _ = child_killer.spawn();
```

Comment:
I think you might want `.status()` here instead of `.spawn()` since we don't want this to be a long-running thing, and may want log reporting if it fails.

Reply:
not much we can do about it if it fails, but okay